hs/src/Reach/IdClient.hs

```diff
@@ -0,0 +1,207 @@
+module Reach.IdClient
```

Comment:
This code isn't actually used anywhere so it is confusing to interpret for me. I think you need to add a trivial command line tool that uses this library and that you write shell script tests over in the server code base. (That's an idea, not a command)

Reply:
sounds good, I'll work on that